### PR TITLE
Recipes for mariadb-connector-python & mariadb-connector-c

### PR DESCRIPTION
**Pull request by d-cunningham**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by d-cunningham** [comment]
@mxr-conda agree to maintain?

File: recipes/mariadb-connector-c/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - cmake
+    - ninja
+  host:
+    - openssl
+  run:
+    - mysql
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/meta.yaml:25]: I suspect you don't need the full server + libs + client here? You can search through the list of mysql components on this page: https://conda-forge.org/feedstock-outputs/index.html

File: recipes/mariadb-connector-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/mariadb-{{ version }}.tar.gz
+  sha256: 3c7eebcb94588b879069a00ae3894ef404f24ed350c40d633ccb98964bc1de55
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: True  # [win or osx]
+  missing_dso_whitelist:
+    - '*'
```
Comment (chrisburr) [inline recipes/mariadb-connector-python/meta.yaml:17]: You probably shouldn't use this. If you actually need it a comment should be included for why.
```suggestion
```

File: recipes/mariadb-connector-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{% set name = "mariadb" %}
+{% set version = "1.0.7" %}
+
+package:
+  name: {{ name|lower }}
```
Comment (chrisburr) [inline recipes/mariadb-connector-python/meta.yaml:5]: Perhaps it'll be less confusing if this package name includes Python:
```suggestion
  name: {{ name|lower }}-python
```

File: recipes/mariadb-connector-c/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/mariadb-corporation/mariadb-connector-c/archive/v{{ version }}.tar.gz
+  sha256: 6da2563a7940457846267989732b97028c350d7de07f89e355f15646641d0ed7
+
+build:
+  number: 0
+  missing_dso_whitelist:
+    - '*'
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/meta.yaml:15]: Same here

File: recipes/mariadb-connector-c/build.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    -DENABLE_FORMAT=OFF \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON \
+    -DWITH_MYSQLCOMPAT=1 \
+    -DINSTALL_LIBDIR=lib \
+    ..
+
+make -k -j${CPU_COUNT} || true
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/build.sh:17]: Why do you have the `|| true`?

File: recipes/mariadb-connector-c/build.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -e -x
+
+mkdir build
+cd build
+
+cmake \
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/build.sh:7]:
```suggestion
cmake ${CMAKE_ARGS} \
```

File: recipes/mariadb-connector-c/bld.bat [hunk 7]
```diff
@@ -0,0 +1,18 @@
+mkdir build && cd build
+
+:: Set INSTALL_DOCREADMEDIR to a junk path to avoid installing the README into PREFIX
+cmake -GNinja ^
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/bld.bat:4]:
```suggestion
cmake %CMAKE_ARGS% ^
      -GNinja ^
```

File: recipes/mariadb-connector-c/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    - ninja
+  commands: 
+    - mysql --help
+    - mysql_config --version
+
+about:
+  home: https://mariadb.com/kb/en/mariadb-connector-c/
+  license: LGPL-2.1-or-later
+  license_family: LGPL
+  license_file: COPYING.LIB
```
Comment (chrisburr) [inline recipes/mariadb-connector-c/meta.yaml:40]: Is this file not available in the source tarball?

File: recipes/mariadb-connector-c/build.sh (hunk 5 above)
Comment (d-cunningham) [reply recipes/mariadb-connector-c/build.sh:17]: Im not sure what `|| true` does. Copied it from another feedstock and forgot to test without it.

File: recipes/mariadb-connector-c/meta.yaml (hunk 4 above)
Comment (d-cunningham) [reply recipes/mariadb-connector-c/meta.yaml]: fails on osx when I remove dso_whitelist due to a OverLinkingError

File: recipes/mariadb-connector-c/meta.yaml (hunk 8 above)
Comment (d-cunningham) [reply recipes/mariadb-connector-c/meta.yaml:40]: which file are you referring too?

**Comment by d-cunningham** [comment]
@chrisburr if you get a chance can you take a look at the recipe revisions? It would be much appreciated. Trying to close some open PR's. Thanks!

**Comment by d-cunningham** [comment]
bump

File: recipes/mariadb-connector-c/meta.yaml (hunk 8 above)
Comment (chrisburr) [reply recipes/mariadb-connector-c/meta.yaml:39]: In that case you don't need to include it in the PR. Conda will find it from the source tarball and it's preferred to do that whenever possible.

**Comment by ProfLeao** [comment]
Hello everyone!
How are things going with mariadb-connector-python?

**Pull request closed without merging**